test/locks.js

```diff
@@ -39,231 +47,219 @@ describe('Locks', function() {
-        // so override that in order to get apostrophe to
-        // listen normally and not try to run a task. -Tom
-        apos.argv._ = [];
-        return callback(null);
-      },
-      afterListen: function(err) {
-        assert(!err);
-        done();
       }
     });
-  });
 
-  it('cleanup', function(done) {
-    apos.locks.db.remove({}, function(err) {
-      assert(!err);
-      done();
-    });
+    assert(apos.modules['apostrophe-locks']);
+    assert(apos.modules['apostrophe-locks-1']);
+    assert(apos.modules['apostrophe-locks-2']);
+    assert(apos.modules['apostrophe-locks-3']);
   });
 
-  it('should allow a single lock without contention uneventfully', function(done) {
-    let locks = apos.modules['apostrophe-locks'];
-    return async.series([ lock, unlock ], function(err) {
-      assert(!err);
-      done();
-    });
-    function lock(callback) {
-      return locks.lock('test', callback);
-    }
-    function unlock(callback) {
-      return locks.unlock('test', callback);
+  it('cleanup', async function() {
+    try {
+      const response = await apos.locks.db.remove({});
+
+      assert(response.result.ok === 1);
+    } catch (e) {
+      assert(false);
     }
   });
 
-  it('should allow two differently-named locks uneventfully', function(done) {
-    let locks = apos.modules['apostrophe-locks'];
-    return async.series([ lock1, lock2, unlock1, unlock2 ], function(err) {
-      assert(!err);
-      done();
-    });
-    function lock1(callback) {
-      return locks.lock('test1', callback);
-    }
-    function unlock1(callback) {
-      return locks.unlock('test1', callback);
-    }
-    function lock2(callback) {
-      return locks.lock('test2', callback);
-    }
-    function unlock2(callback) {
-      return locks.unlock('test2', callback);
+  it('should allow a single lock without contention uneventfully', async function() {
+    const locks = apos.modules['apostrophe-locks'];
+
+    try {
+      await locks.lock('test');
+
+      await locks.unlock('test');
+    } catch (e) {
+      assert(false);
     }
+
+    // function lock(callback) {
+    //   return locks.lock('test', callback);
+    // }
+    // function unlock(callback) {
+    //   return locks.unlock('test', callback);
+    // }
   });
 
-  it('should flunk a second lock by the same module', function(done) {
-    let locks = apos.modules['apostrophe-locks'];
-    return async.series([ lock, lockAgain, unlock, unlockAgain ], function(err) {
-      assert(!err);
-      done();
-    });
-    function lock(callback) {
-      return locks.lock('test', callback);
-    }
-    function lockAgain(callback) {
-      return locks.lock('test', function(err) {
-        // SHOULD fail
-        assert(err);
-        return callback(null);
-      });
+  it('should allow two differently-named locks uneventfully', async function() {
+    const locks = apos.modules['apostrophe-locks'];
+
+    try {
+      await locks.lock('test1');
```

Comment:
look at the async.series call you're replacing... the idea here is to lock both, then unlock both (both locks exist simultaneously).